File: leetcode/334-increasing-triplet-subsequence/increasing-triplet-subsequence.cpp
class Solution {
public:
    bool increasingTriplet(vector<int>& nums) {
       if(nums.size()<3){
          return false;
       } 
       int l=nums.size();
       int min1=INT_MAX;
       int min2=INT_MAX;
       for(int i=0;i<l;i++){
        if(nums[i]<=min1){
            min1=nums[i];
        }
        else if(nums[i]<=min2){
            min2=nums[i];
        }
        else{
            return true;
        }
       } 
       return false;

    }
};

File: leetcode/334-increasing-triplet-subsequence/increasing-triplet-subsequence.py
class Solution(object):
    def increasingTriplet(self, nums):
        """
        :type nums: List[int]
        :rtype: bool
        """
        num_i = num_j = float('inf')
        for num in nums:
            if num <= num_i:
                num_i = num
            elif num <= num_j:
                num_j = num
            else:
                return True
        return False